public/js/core.js

```diff
@@ -52,44 +53,128 @@ weSizeItApp.factory('socket', ['$rootScope', function ($rootScope) {
     };
 }]);
 
-weSizeItApp.config(['$routeProvider', function ($routeProvider) {
-    $routeProvider
-    .when('/login', {
-        templateUrl: 'views/login.html',
-        public: true,
-        login: true,
-    })
-    .when('/signup', {
-        templateUrl: 'views/signup.html',
-        public: true,
-    })
-    .when('/room', {
-        templateUrl: 'views/room.html',
-        controller: 'mainViewCtrl',
-    })
-    .otherwise({
-        redirectTo: '/room',
-    });
-}]);
+weSizeItApp.config(function($stateProvider, $urlRouterProvider, $locationProvider) {
+    
+    $urlRouterProvider.otherwise('/home');
+    
+    $locationProvider.html5Mode(true);
+    
+    $stateProvider
+        
```

Comment:
I suggest you add a parent application state, then make all the other states substates of it.

This parent state can fill in the "nav" and the substates can override it, but only if needed.

```
        .state('app', {
             views: {
                 nav: {
                   templateUrl: 'views/general-bar.html'
                 },
                 content: {
                   template: '<div ui-view></div>' // nested placeholder for nested states to fill in
                 }
             }
         })
```

``` 
   // simpler configuration for each state that doesn't need to update the nav
   // because they use the default unnamed view
  .state('home', {
             parent: 'app', // Make this state a substate of 'app'
             url: '/home',
             templateUrl: 'views/landing.html'
            }
        })
```

```
       // states that need to modify the 'nav' can still use the 'named views' config style
         .state('register', {
             url: '/register',
             views: {
                 "nav@": { // nav view at root state
                   templateUrl: 'views/general-bar.html'
                 },
                 "@app": { // unnamed view at app state
                   templateUrl: 'views/register.html'
                 }
             }
         })
```